fenix/app/src/main/java/org/mozilla/fenix/browser/readermode/BrowserNavigator.kt

```diff
@@ -0,0 +1,173 @@
+package org.mozilla.fenix.browser.readermode
+
+import androidx.annotation.IdRes
+import androidx.navigation.NavController
+import androidx.navigation.NavDirections
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.search.SearchEngine
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.base.profiler.Profiler
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.storage.HistoryMetadataKey
+import mozilla.components.feature.search.SearchUseCases
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.feature.tabs.TabsUseCases
+import mozilla.components.support.ktx.kotlin.isUrl
+import mozilla.components.support.ktx.kotlin.toNormalizedUrl
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.Components
+import org.mozilla.fenix.ext.alreadyOnDestination
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.utils.Settings
+
+class BrowserNavigator(
+    private val addTabUseCase: TabsUseCases.AddNewTabUseCase,
+    private val sessionUseCases: SessionUseCases,
+    private val searchUseCases: SearchUseCases,
+    private val appStore: AppStore,
+    private val browserStore: BrowserStore,
+    private val profiler: Profiler?,
+    private val settings: Settings,
+    private val getNavDirections: (BrowserDirection, String?) -> NavDirections?,
+    private val getNavController: () -> NavController,
```

Comment:
why a lambda instead of just the actual nav controller? so `BrowserNavigator` doesn't hold onto the reference?